File: platforms/android/devkit/devkit/src/main/java/io/nativeblocks/devkit/feature/preview/domain/repository/ParameterRepository.kt
package io.nativeblocks.devkit.feature.preview.domain.repository

import io.nativeblocks.devkit.feature.preview.domain.model.Parameter

internal interface ParameterRepository {
    fun getAll(): List<Parameter>
    fun add(key: String, value: String): Parameter
    fun update(id: String, key: String, value: String)
    fun delete(id: String)
    fun clear()
    fun toMap(): Map<String, String>
}

File: platforms/ios/devkit/Sources/NativeblocksDevkit/feature/preview/domain/repository/ParameterRepository.swift
import Foundation

internal protocol ParameterRepository {
    func getAll() -> [PreviewParameter]
    func add(key: String, value: String) -> PreviewParameter
    func update(id: String, key: String, value: String)
    func delete(id: String)
    func clear()
    func toMap() -> [String: String]
}
